terra_notebook_utils/cli/__init__.py

```diff
@@ -45,9 +45,9 @@ def write(cls):
     @classmethod
     def resolve(cls, override_workspace: Optional[str], override_namespace: Optional[str]) -> Tuple[str, str]:
         workspace = override_workspace or (cls.info['workspace'] or WORKSPACE_NAME)
-        namespace = override_namespace or (cls.info['workspace_namespace'] or WORKSPACE_GOOGLE_PROJECT)
+        namespace = override_namespace or (cls.info['workspace_namespace'] or WORKSPACE_NAMESPACE)
```

Comment:
We should probably still account for `WORKSPACE_GOOGLE_PROJECT`:

```
namespace = override_namespace or (cls.info['workspace_namespace'] or WORKSPACE_NAMESPACE or WORKSPACE_GOOGLE_PROJECT)
```

Especially since TNU can be used outside of Terra, and outside of Terra, `WORKSPACE_NAMESPACE` may not exist if you aren't tying yourself to a workspace but still want to pay with a billing account.

Reply:
I agree, we should make it easier to configure a billing project. Currently you can pass the API parameter directly, or set the environment variable(s).

I wanted to refactor parts of how the CLIConfig worked, and make it easier to add more variables, like `WORKSPACE_GOOGLE_PROJECT`. To avoid scope creep, I left it out of this PR, hoping that the current configuration options are good enough for now.

What do you think?

Reply:
@DailyDreaming, I agree that it would be good to treat configuration and override of all three variables (workspace namespace, workspace name, and google project) consistently. I realize you are working without the benefit of the detailed review I provided to @jessebrennan regarding the PPW changes to these variables and how they are used. The suggestion you provided would not provide the effect you intend.
Also, as you say, it is important that TNU work well outside of Terra, yet in all the cases where the google project is needed (for data access), the workspace_namespace would also be needed for the call to enable requester pays.

@jessebrennan I agree with both your points above. For consistency, CLIConfig should be extended to support all three variables. Conceptually it fits within the scope of this PR, yet there are a lot of code touchpoints in this PR already that function properly as they stand, so I am fine with deferring the CLIConfig change to a subsequent PR.

Reply:
@jessebrennan @mikebaumann I was under the impression that workspaces created before a certain date should still have the same behavior and defaults, including the final default of `WORKSPACE_GOOGLE_PROJECT`.  But this default looks like it was never used, as even our test workspace has:

```
WORKSPACE_NAMESPACE=firecloud-cgl
GOOGLE_PROJECT=firecloud-cgl
```

So I suspect `WORKSPACE_GOOGLE_PROJECT` never is and probably never was actually reached as a default.  I believe my misunderstanding was not with the new PPW changes, but that this default was actually needed for the old functionality.  I'm fine with discarding this default now.

Reply:
Thanks for the clarification/elaboration, @DailyDreaming.
Yes, essentially, my understanding is that `WORKSPACE_GOOGLE_PROJECT` is a TNU-internal name representing the `GOOGLE_PROJECT` environment variable. I don't think there ever was a `WORKSPACE_GOOGLE_PROJECT` environment variable.

Reply:
@mbaumann-broad Yes, `WORKSPACE_GOOGLE_PROJECT` is not an environment variable, it is a python variable.  The python variable that was set to the env var `GOOGLE_PROJECT` on that python file line could never be reached as a default in Terra.  You could artificially manipulate your environment variables to have the env var `WORKSPACE_NAMESPACE` but not the env var `GOOGLE_PROJECT` but this would never happen in Terra, since both of these are set.  I hope that that clarifies things.